eclair-core/src/test/scala/fr/acinq/eclair/io/PeerConnectionSpec.scala

```diff
@@ -372,5 +373,49 @@ class PeerConnectionSpec extends TestKitBaseClass with FixtureAnyFunSuiteLike wi
     assert(new String(warn2.data.toArray).startsWith("invalid announcement sig"))
   }
 
+  test("establish transient connection") { f =>
+    import f._
+    connect(nodeParams, remoteNodeId, switchboard, router, connection, transport, peerConnection, peer, isPersistent = false)
+    val probe = TestProbe()
+    val (_, message) = buildMessage(randomKey(), randomKey(), Nil, Left(Recipient(remoteNodeId, None)), Nil)
+    probe.send(peerConnection, message)
+    probe watch peerConnection
+    Thread.sleep(900)
+    assert(peerConnection.stateName === PeerConnection.CONNECTED)
+    Thread.sleep(200)
+    probe.expectTerminated(peerConnection, max = Duration.Zero)
+  }
+
+  test("keep using transient connection") { f =>
```

Comment:
This test is very hard to write without `Thread.sleep`, I simply don't think it's worth the effort.

Your current version risks being very flaky, because it depends on timing, and during CI where a lot of tests are run in parallel, the execution of this specific test may pause for a while which could make it fail. I would simply remove that test, unless you find a clever way of testing this in a non-blocking way?